salt/modules/network.py

```diff
@@ -1522,7 +1522,5 @@ def iphexval(ip):
         salt '*' network.iphexval 10.0.0.1
     '''
     a = ip.split('.')
-    hexval = ""
-    for val in a:
-        hexval = ''.join([hexval, hex(int(val))[2:].zfill(2)])
-    return hexval.upper()
+    hexval = ['%02X' % int(x) for x in a]
```

Comment:
This may get flagged by pylint if we are still checking for E1321. If that happens, just change this line to:

```python
hexval = ['%02X' % int(x) for x in a]  # pylint: disable=E1321
```

**NOTE:** 2 spaces after the end of the list comprehension.